unix/main.c

```diff
@@ -416,7 +416,7 @@ int main(int argc, char **argv) {
             } else if (strcmp(argv[a], "-v") == 0) {
                 mp_verbose_flag++;
             } else if (strncmp(argv[a], "-O", 2) == 0) {
-                if (isdigit(argv[a][2])) {
+                if (isdigit((byte)argv[a][2])) {
```

Comment:
Well if the standard says 'undefined behaviour' the implementation is free to do whatever it wants, including kabooming all over the place. So in theory an implementation indexing a table without checking arguments is perfectly fine. Though I agree that in practice however it could just check the argument, and most definitely it cannot raise bogus implementation-defined warnings.